thor/src/lib.rs

```diff
@@ -144,22 +143,113 @@ impl Channel {
     /// Update the distribution of coins in the channel.
     ///
     /// It assumes that the counterparty has already agreed to update the
-    /// channel with the `new_balance` and the same `timelock` and will call
-    /// the same API (or an equivalent one).
+    /// channel with the same balance and `timelock` and will call the same API
+    /// (or an equivalent one).
     ///
     /// Consumers should implement the traits `SendMessage` and `ReceiveMessage`
     /// on the `transport` they provide, allowing the parties to communicate
     /// with each other.
     pub async fn update<T>(
         &mut self,
         transport: &mut T,
-        new_balance: Balance,
+        Balance { ours, theirs }: Balance,
+        time_lock: u32,
+    ) -> anyhow::Result<()>
+    where
+        T: SendMessage + ReceiveMessage,
+    {
+        let split_output_self = SplitOutput::Balance {
+            amount: ours,
+            address: self.final_address_self.clone(),
+        };
+
+        let split_output_other = SplitOutput::Balance {
+            amount: theirs,
+            address: self.final_address_other.clone(),
+        };
+
+        self._update(
+            transport,
+            vec![split_output_self, split_output_other],
+            time_lock,
+        )
+        .await
+    }
+
+    /// Perform an atomic swap with a thor channel as beta ledger in the
+    /// role of Alice.
+    pub async fn add_ptlc_alice<T>(
+        &mut self,
+        _transport: &mut T,
+        _new_split_outputs: Vec<SplitOutput>,
+        _encryption_secret_key: Scalar,
+        _alpha_absolute_expiry: u32,
+        _TX_s_time_lock: u32,
+        _ptlc_redeem_time_lock: u32,
+    ) -> anyhow::Result<()>
+    where
+        T: SendMessage + ReceiveMessage,
+    {
+        // TODO: Think about how to handle the three expiries.
```

Comment:
If I thought it through correctly than 1 relative and 2 absolute would work well.

- `t_0`- relative timeout for `Tx_s`
- `t_1`- absolute timeout for `PTLC` on `Tx_s` (beta ledger)
- `t_2`- absolute timeout for `PTLC` on-chain (alpha ledger)

for example
* `start`= time when the swap started
* `now()` = function to get current date
* `t_0`=  144 blocks or 24h
* `t_1`= `start+24h+12h`
* `t_2`= `start+24h+24h`

* Bob should initiate refund (i.e. close the channel) if `now()` is getting close  to (or just past) `start+12h`
  * if he does that 2 things can happen: (i) `t_1` has passed, and he refunds immediately, or (ii) `t_1` has not passed and Alice can still redeem. If Alice redeems (even after `t_1` had passed) Bob still has time to redeem alpha ledger
* If Alice reveals the secret and Bob does not agree on updating the channel, Alice should settle before `now()` reaches `start+12h` (assuming she wants to redeem). If not, she can just wait until `t_2` expires.

Reply:
I modified the TODO you referenced linking to your comment so that we can consider this (which seems correct after a quick look) when tackling the TODO.